lib/systems/default.nix

```diff
@@ -76,6 +77,44 @@ let
     # TODO: deprecate args.rustc in favour of args.rust after 23.05 is EOL.
     rust = args.rust or args.rustc or {};
 
+    toolchain = {
+      cc = if final.useLLVM || final.isDarwin then "clang"
+        else if final.useArocc then "arocc"
+        else if final.useZig then "zig"
+        else "gcc";
+
+      bintools = if final.useLLVM || final.useArocc || final.useZig || final.isDarwin then
+        "llvm"
+      else "gnu";
+
+      cxxlib =
+        /**/ if final.useLLVM || final.useArocc || final.useZig || final.isDarwin then "libcxx"
+        else "libstdcxx";
```

Comment:
I think this needs a `cxxrtlib` to go with it, per https://github.com/NixOS/nixpkgs/pull/365057#discussion_r2009312157.

Reply:
Alright, should we imply the `cxxrtlib` based on the `cxxlib` or should it be based on the "toolchain" in use?